client/state/experiments/selectors.ts

```diff
@@ -0,0 +1,40 @@
+/**
+ * External Dependencies
+ */
+import { get } from 'lodash';
+
+/**
+ * Internal Dependencies
+ */
+import { AppState } from 'types';
+
+/**
+ * Returns the user's assigned variation for a given experiment
+ *
+ * @param state The application state
+ * @param experiment The name of the experiment
+ */
+export const getVariationForUser = ( state: AppState, experiment: string ) =>
+	get( state, [ 'experiments', 'tests', experiment ], null );
+
+/**
+ * Returns true if the variations are loading for the current user
+ *
+ * @param state The application state
+ */
+export const isLoading = ( state: AppState ) => get( state, [ 'experiments', 'isLoading' ], true );
+
+/**
+ * Get's the anon id for the user, if set
+ *
+ * @param state The application state
+ */
+export const getAnonId = ( state: AppState ) => get( state, [ 'experiments', 'anonId' ], null );
```

Comment:
Do we need to expose the `anonId` from the state? It seems like an internal detail to me, but I don't have a good mental image of how everything connects.

Also, `Get's` -> `Gets` :slightly_smiling_face:

Reply:
> Do we need to expose the anonId from the state?

It appears the cookie gets deleted at some point... so we can't rely on it sticking around. We need it in case someone hits a landing page with an anon-experiment and gets assigned to that experiment. When they login, we need to know what they were assigned to as an anon-user so we need the value of the cookie to send to the assignment API to let it know that this user was that anon-user.

Reply:
Make sense. I was thinking more about whether we should export this function as opposed to using the anon ID internally and still storing it in the state.

Another thing I thought about is the order of execution: Is it possible for our code to run before the code that sets the cookie? Looks like it sometimes happens on the client side:

https://github.com/Automattic/wp-calypso/blob/bcddb6c2e3e9c983e7dccb8acaff5da784ad2b65/client/lib/analytics/index.js#L118-L127

However, from poking around it looks like there's also server-side code to set the same cookie. Thoughts?

Reply:
Maybe we should make a new cookie instead of fighting over the tracks anon cookie?

![standards](https://imgs.xkcd.com/comics/standards.png)

(I don't think that's a bad idea btw.) However, we should probably recheck it before making the API request.

Reply:
We need to use a consistent anonId to match our event tracking. It looks like initialising the `analytics` module would ensure that the cookie exists, but is there a better way to ensure it happens?

Taking a step back, is the only purpose of saving the anonId to pass it to an API call? If so, can't we assume that the server knows the anonId? If we can't make this assumption, how would the server track assignment for anons? Rely on the passed anonId?

Reply:
> It looks like initialising the `analytics` module would ensure that the cookie exists, but is there a better way to ensure it happens?

Actually, I don't think the `analytics` module will provide any guarantee that the cookie will exist. We typically depend on Tracks' `w.js` to serializes the `_ui` cookie value.

I think following @yanirs' advice [here](https://github.com/Automattic/wp-calypso/pull/38907/files#r368739892) makes the most sense. 

Pinging @bperson and @xyu for guidance.

Reply:
> Actually, I don't think the `analytics` module will provide any guarantee that the cookie will exist.

Why not? `analytics.initialize()` calls `checkForBlockedTracks()`, which calls the code I referenced, though it does happen in the `catch()` of `_loadTracksResult` so I'm not sure about the order of execution.

Reply:
I personally don't know anything after `w.js`. I can try to understand what Calypso' analytics module tries to do if needed, but I'd be starting from scratch 🤷‍♂

Reply:
I've verified that it doesn't capture the anon-id from a fresh boot of calypso and an anon user. I'll think on this and get back to you.

Reply:
OK, after a few more tests, it does capture the anonid WHEN the user logs in, but not before.

Thinking about it a bit more, this is what I'm going to do:

1. Continue to capture the anonid on app init. This will capture it if they come from a landing page or something.
2. Capture the anonid if we haven't already, when we request from the API.

Reply:
> will provide any guarantee that the cookie will exist.

There's no guarantee the cookie will exist (it gets deleted when the user is aliased as far as I can tell). All we can guarantee is "best effort".

Reply:
( thinking more about this )

> It appears the cookie gets deleted at some point...

If it can be useful: this most likely happens in `w.js` when the user gets identified with `identifyUser` ( but I'm not sure exactly when that happens anymore in Calypso' flows )

Reply:
> > Actually, I don't think the `analytics` module will provide any guarantee that the cookie will exist.
> 
> Why not? `analytics.initialize()` calls `checkForBlockedTracks()`, which calls the code I referenced, though it does happen in the `catch()` of `_loadTracksResult` so I'm not sure about the order of execution. @yanirs 

`checkForBlockedTracks` will only initialize the anonymous ID if the `loadScript( '//stats.wp.com/w.js?60' )` fails. 

Even if the network request does fail, it'd be too late; the analytics module is initialized as [a middleware](https://github.com/Automattic/wp-calypso/blob/master/client/boot/common.js#L248) after the [Redux store has already been initialized](https://github.com/Automattic/wp-calypso/blob/master/client/boot/app.js#L32).

FWIW, landing pages use [a different boot logic](https://github.com/Automattic/wp-calypso/blob/master/client/landing/login/index.js#L35).